src/cpu/Dasm.cc

```diff
@@ -23,96 +23,122 @@ void appendAddrAsHex(std::string& output, uint16_t addr)
 	strAppend(output, '#', hex_string<4>(addr));
 }
 
-unsigned dasm(const MSXCPUInterface& interface, uint16_t pc, std::span<uint8_t, 4> buf,
-              std::string& dest, EmuTime::param time,
-              function_ref<void(std::string&, uint16_t)> appendAddr)
+unsigned instructionLength(std::span<const uint8_t> bin)
+{
+	if (bin.empty()) return 0;
+	auto t = instr_len_tab[bin[0]];
+	if (t < 4) return t;
+
+	if (bin.size() < 2) return 0;
+	return instr_len_tab[64 * t + bin[1]];
+}
```

Comment:
Would it be cleaner if this function returns a `std::optional<unsigned>` instead of using value `0` to indicate an error?
* That better separates the valid values from the error case ("could not calculate instruction length")
* Callers are less likely to forget to check for this error.

Or alternatively you could indicate errors with `unsigned(-1)`. That way `bin.size() < len` is sufficient to check both errors "could not calculate instruction length" and "blob size too small for instruction".

My personal preference is `std::optional`.